homeassistant/components/samsungtv/__init__.py

```diff
@@ -135,6 +135,23 @@ async def async_setup_entry(hass: HomeAssistant, entry: SamsungTVConfigEntry) ->
             )
     bridge = await _async_create_bridge_with_updated_data(hass, entry)
 
+    @callback
+    def _access_denied() -> None:
+        """Access denied callback."""
+        LOGGER.debug("Access denied in getting remote object")
+        hass.create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={
+                    "source": SOURCE_REAUTH,
+                    "entry_id": entry.entry_id,
+                },
+                data=entry.data,
+            )
+        )
+
+    bridge.register_reauth_callback(_access_denied)
```

Comment:
The reauth callback isn't always called from async context but sometimes from the executor. That's not allowed, since we pass an async callback here.

Reply:
I don't think it's ever called from the executor.

I think that:
- the "legacy" bridge runs on the executor, but never calls the reauth callback.
- and the "regular" bridge runs always on the async context.

Reply:
https://github.com/home-assistant/core/blob/cb502263fd78743717ae540ab2b91412bfb91c05/homeassistant/components/samsungtv/bridge.py#L328-L374

Line 354 calls the executor.